pkg/controller/builder.go

```diff
@@ -253,23 +254,61 @@ func UpdateDeploymentProxyHook(pc clients.ProxyClientInterface) deploymentcontro
 	}
 }
 
-func setContainerArg(con *corev1.Container, arg, value string) {
+// The return behavior:
+// 1. If the input arg value matches the container argument = success
+// 2. If the input arg value does not match the container argument = error
+func setContainerArg(con *corev1.Container, arg, value string) error {
 	// Need to remove any existing `arg` arguments first
 	// This could happen because the experimental manifest may have feature gates already enabled
 	var args []string
+	var containerValues []string
 	log := klog.FromContext(context.Background()).WithName("builder")
 
+	// Calculate the container argument value and create a new container argument list without it
 	for _, a := range con.Args {
 		if !strings.HasPrefix(a, arg) {
 			args = append(args, a)
 		} else {
-			log.Info("Removing arg", "container", con.Name, "argument", a)
+			// Save off original argument value by removing prefix
+			s := strings.TrimPrefix(a, arg)
+			s = strings.TrimPrefix(s, "=")
+			log.Info("Removing argument", "container", con.Name, "arg", arg, "value", s)
+			if s != "" {
+				containerValues = append(containerValues, strings.Split(s, ",")...)
+			}
 		}
 	}
 
+	containerValue := ""
+	if len(containerValues) > 0 {
+		sort.Strings(containerValues)
+		containerValue = strings.Join(containerValues, ",")
+	} else {
+		// Otherweise, there were no original arguments, so just set what was passed in
+		log.Info("No existing arguments of type", "arg", arg)
+	}
+
+	// Check to see if the container values match the new values
+	if value != containerValue {
+		// Arguments do not match, so return an error.
+		// The likely reason for this error is that the feature flag coding has not completed yet;
+		// either the cluster-olm-operator mapping is not present,
+		// or the feature was not added to the experimental manifest.
+		err := fmt.Errorf("new and existing arguments do not match")
+		log.Error(err, "New and existing arguments do not match", "arg", arg, "new", value, "existing", containerValue)
+		return err
+	}
+
+	// Otherwise they match, so just substitute the new arguments
+	log.Info("Existing arguments match new arguments", "arg", arg, "value", value)
+
+	// Update the args, this will make a previously multiline argument into a single line.
 	if value != "" {
 		log.Info("Updated args", "container", con.Name, "arg", arg, "value", value)
 		args = append(args, fmt.Sprintf("%s=%s", arg, value))
 	}
+
+	// Update the containers arguments
 	con.Args = args
+	return nil
```

Comment:
It seems like this function could be simplified quite a bit by:
1. Loop through the container args
2. For each arg, use `strings.Cut(arg, "=")` to split into `name` and `value`
3. If the `name` doesn't match the passed in `arg`, just `continue`
4. Use `strings.Join(slices.Sorted(slices.Values(strings.Split(val, ","))), ",")` to construct comparable existing and desired values
5. If they don't match return the error; if they do, note that we found the arg
6. After the loop, if we have not found the arg, add it to con.Args

Reply:
Ah, now I see the nuance of handling multiple `--feature-gates` arguments, and needing to collect their values. I do still wonder if we could simplify it some though.

Reply:
Another couple of thoughts:
- the function name and the actual function implementation seem to have drifted
- is this functionality still actually applicable to any arbitrary argument that might show up in the container args? i feel like it is different enough where we may want to specifically look for `--feature-gates`  args and handle those separately.

Reply:
But yes, this is much more specific for `--feature-flags", and we don't process any other flags. And due to specialized behavior, yeah, we might want to rename it.

Reply:
I am not finding `List` in `sets`... (i.e. https://pkg.go.dev/k8s.io/apimachinery/pkg/util/sets)
```
	haveValues := strings.Join(sets.List(foundValues), ",")
```
Unfortunately there's `UnsortedList()` (but not `List()` or `SortedList()`, so it still requires sorting. And `slices.Sorted()` wants some type of iterator, so I can't use the output of `UnsortedList()` directly...

But I think I got it...

Reply:
For future reference, `set.List` is here: https://github.com/kubernetes/apimachinery/blob/v0.33.3/pkg/util/sets/set.go#L201

But what you have now is effectively the same, and is symmetric with computing `wantValues`, so looks good to me.